core/src/main/java/org/elasticsearch/node/NodeModule.java

```diff
@@ -58,5 +97,30 @@ protected void configure() {
         bind(Node.class).toInstance(node);
         bind(MonitorService.class).toInstance(monitorService);
         bind(NodeService.class).asEagerSingleton();
+        bind(ProcessorsRegistry.class).toInstance(processorsRegistry);
+    }
+
+    /**
+     * Returns the node
+     */
+    public Node getNode() {
+        return node;
+    }
+
+    /**
+     * Adds a processor factory under a specific type name.
+     */
+    public void registerProcessor(String type, Function<TemplateService, Processor.Factory<?>> processorFactoryProvider) {
+        processorsRegistry.registerProcessor(type, processorFactoryProvider);
+    }
+
+    public static boolean isNodeIngestEnabled(Settings settings) {
+        return settings.getAsBoolean("node.ingest", true);
+    }
+
+    public static boolean isNodeIngestEnabled(ImmutableOpenMap<String, String> nodeAttributes) {
+        String ingestEnabled = nodeAttributes.get("ingest");
+        //reproduces same logic used in settings.getAsBoolean used above
+        return Booleans.parseBoolean(ingestEnabled, true);
```

Comment:
`node.ingest` should not be a string here but this:

```
public static final Setting<Boolean> NODE_INGEST_SETTTING = Setting.getBoolean("node.ingest", true, false, Scope.CLUSTER);
```

then register it in `ClusterSettings#BUILT_IN_CLUSTER_SETTINGS` and use is here
 ```Java
public static boolean isNodeIngestEnabled(Settings settings) {
  return NODE_INGEST_SETTTING.get(settings);
}

public static boolean isNodeIngestEnabled(ImmutableOpenMap<String, String> nodeAttributes) {
 String ingestEnabled = nodeAttributes.get("ingest");
  return ingestEnabled == null ? false : isNodeIngestEnabled(Settings.builder().put(ingestEnabled)); // if this is too heavy we can add a util method to Setting.java
}
```

the new setting is not lenient!